common/service/config/config.go

```diff
@@ -283,16 +282,26 @@ type (
 
 	// Archival contains the config for archival
 	Archival struct {
-		// Status is the status of archival either: enabled, disabled, or paused
+		// History is the config for the history archival
+		History HistoryArchival `yaml:"historyArchival"`
+		// Visibility is the config for visibility archival
+		Visibility VisibilityArchival `yaml:"visibilityArchival"`
+	}
+
+	// HistoryArchival contains the config for history archival
+	HistoryArchival struct {
+		// Status is the status of history archival either: enabled, disabled, or paused
 		Status string `yaml:"status"`
 		// EnableReadFromArchival whether history can be read from archival
 		EnableReadFromArchival bool `yaml:"enableReadFromArchival"`
-		// DefaultBucket is the default bucket used for archival in case domain does not specify override
-		DefaultBucket string `yaml:"defaultBucket"`
-		// Filestore the configuration for file based blobstore
-		Filestore *filestore.Config `yaml:"filestore"`
-		// S3store the configuration for amazon s3 based blobstore
-		S3store *s3store.Config `yaml:"s3store"`
+	}
+
+	// VisibilityArchival contains the config for visibility archival
+	VisibilityArchival struct {
```

Comment:
Although this struct currently contains the same fields as the HistoryArichival struct above, they will be different as there will be different implementation for history and visibility archival and their configs (will be new field in these structs) are different.